pkg/types/validation/installconfig.go

```diff
@@ -699,3 +703,35 @@ func validateFIPSconfig(c *types.InstallConfig) field.ErrorList {
 	}
 	return allErrs
 }
+
+// validateCapabilities checks if additional, optional openshift components are specified in the
+// install-config to be included in the installation
+func validateCapabilities(c *types.Capabilities, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if c.BaselineCapabilitySet == "" {
+		allErrs = append(allErrs, field.Required(fldPath.Child("baselineCapabilitySet"), "baseline capability set required"))
+
+		return allErrs
+	}
+	// Check to see if the additionalEnabledCapabilities is defined in openshift/api
+	if availableCapabilities, found := configv1.ClusterVersionCapabilitySets[c.BaselineCapabilitySet]; found {
+		// create a set of the available capability set
+		availableCapabilitySet := sets.NewString()
+		for _, capability := range availableCapabilities {
```

Comment:
`additionalEnabledCapabilities` is expected to contain caps that are not in the baseline set.  Some cases:

* An `additionalEnabledCapabilities` cap is in the baseline set.  This is redundant stutter, because the set already enables the capability.  The user could drop the cap from `additionalEnabledCapabilities` and get exactly the same set of enabled caps.  But still, redundant isn't broken, so probably not worth pestering users about these.
* An `additionalEnabledCapabilities` cap is not in the baseline set, but it is in some other set.  This is great, the user is extending their requested cap set beyond their selected base set.  Lovely.
* An `additionalEnabledCapabilities` cap is not in any set.  Probably a typo, and we should reject the install-config.

So instead of building `availableCapabilitySet` from the selected baseline set, we should build it to include all caps from every set while iterating over every set in `configv1.ClusterVersionCapabilitySets`.  And then we can complain about unrecognized caps here.

Reply:
I agree with you that this logic is broken. I'll work on a fix. Looking at the ClusterVersionCapabilitySets, it would be nice to maintain a *everything included* set which we could use to look up and figure out whether the user input is kosher or not. WDYT?

Reply:
Also, wouldn't you think that a union of all the sets available in `configv1.ClusterVersionCapabilitySets` could potentially leave out an list of actual optional caps? Hence my mention of the "all possible capabilities" set.

Reply:
I agree that it's theoretically possible to have a capability that belongs to no sets.  But I would be very surprised if we ever had a capability like that, and even if we did, the folks testing that capability would notice when the installer complained about the unrecognized value, and we could patch things up at that point.

But today we do not have any capabilities like that.  In fact, `vCurrent` will contain the complete set of caps for 4.10, so the super simple approach is just to use `vCurrent` to get the valid caps.  Iterating over all sets in the map gives you some protection against some hypothetical future where `vCurrent` leaves out some new capability and a new `allTheCaps` set or whatever takes over that role.

Reply:
Fixed in a10be333d